language/move-lang/src/cfgir/translate.rs

```diff
@@ -291,19 +281,19 @@ fn constant_(
     let fake_acquires = BTreeMap::new();
     let fake_infinite_loop_starts = BTreeSet::new();
     cfgir::refine_inference_and_verify(
-        &mut fake_errors,
+        context.env,
         &context.struct_declared_abilities,
         &fake_signature,
         &fake_acquires,
         &locals,
         &mut cfg,
         &fake_infinite_loop_starts,
     );
-    assert!(fake_errors.is_empty(), "{}", ICE_MSG);
+    assert!(has_prev_errors == context.env.has_errors(), "{}", ICE_MSG);
```

Comment:
@tnowacki Let me know if this change looks good to you or not.

Reply:
Hm.... I think we need an `error_count` for this unfortunately. The check needs to be that compiling constants didn't result in errors. Very bad things happen if that happens.